packer/linux/conf/buildkite-agent/scripts/fix-buildkite-agent-builds-permissions

```diff
@@ -71,15 +71,30 @@ exit_if_blank "${AGENT_DIR}"
 exit_if_blank "${ORG_DIR}"
 exit_if_blank "${PIPELINE_DIR}"
 
-# If we make it here, we're safe to go!
-
 # We know the builds path:
 BUILDS_PATH="/var/lib/buildkite-agent/builds"
 
 # And now we can reconstruct the full agent builds path:
 PIPELINE_PATH="${BUILDS_PATH}/${AGENT_DIR}/${ORG_DIR}/${PIPELINE_DIR}"
 # => "/var/lib/buildkite-agent/builds/my-agent-1/my-org/my-pipeline"
 
-if [[ -e "${PIPELINE_PATH}" ]]; then
-	/bin/chown -R buildkite-agent:buildkite-agent "${PIPELINE_PATH}"
+# If it doesn't exist, then we won't do anything.
+if [[ ! -e "${PIPELINE_PATH}" ]]; then
+	exit 0
+fi
+
+
+# Check for symlink shenanigans
+if [[ "$(realpath "${PIPELINE_PATH}")" != "${PIPELINE_PATH}" ]]; then
```

Comment:
Why not
```suggestion
if [[ -L "${PIPELINE_PATH}" ]]; then
```

Reply:
That catches the final path segment being a symlink, but not the intermediate segments.